src/Tizen.Uix.VoiceControlManager/Tizen.Uix.VoiceControlManager/ExceptionFactory.cs

```diff
@@ -0,0 +1,148 @@
+﻿/*
+* Copyright (c) 2019 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using static Interop.VoiceControlManager;
+
+namespace Tizen.Uix.VoiceControlManager
+{
+    internal static class ExceptionFactory
+    {
+        internal static Exception CreateException(ErrorCode err)
+        {
+            Tizen.Log.Error(LogTag, "Error " + err);
+            Exception exp;
+            switch (err)
+            {
+                case ErrorCode.OutOfMemory:
+                    {
+                        exp = new OutOfMemoryException("Out Of Memory");
```

Comment:
Do not throw Out of memory exception, it only can throw on clr

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types#outofmemoryexception